pkg/kubelet/volume/volume_manager.go

```diff
@@ -185,10 +187,17 @@ type volumeManager struct {
 	// desiredStateOfWorldPopulator runs an asynchronous periodic loop to
 	// populate the desiredStateOfWorld using the kubelet PodManager.
 	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+
+	// mountedVolulmeToClean is the list of volumes already mounted before
+	// kubelet starts. They need to be cleaned up before reconciler runs
+	mountedVolumeToClean []operationexecutor.MountedVolume
 }
 
 func (vm *volumeManager) Run(stopCh <-chan struct{}) {
 	defer runtime.HandleCrash()
+	glog.Infof("Clean up already mounted volumes before Kubelet starts")
+	vm.cleanUpMountedVolumes()
+
```

Comment:
Another question: this PR attempts to do a one time cleanup before starting the volume manager. If kubelet fails to get the list of volumes due to some transient error, the volumes will never be cleaned up.

Also, my [comment](https://github.com/kubernetes/kubernetes/pull/27970/files#r68328317) above is also a problem for the one-time cleanup.